### PR TITLE
Adding first stab on workflow patterns

### PR DESCRIPTION
**Pull request by berndruecker**
This is related to our positioning focusing on advanced workflow patterns more - we want to have one referencable resource to show what those patterns are (and then probably do more specific examples comparing it to other workflow languages)

### CODE REVIEW

**Comment by berndruecker** [comment]
@akeller: I would be ready to push this first version into the docs if you are happy with it (it would live under Components > Concepts). We still need an editorial edit for sure.

grafik

Given that it is hard to review on markdown without the BPMN Models being shown, I would ping some folks to give feedback once it is in the docs (and iterate on the content). 

Would love to get it out within this or next week - as I have a blog post based on this, which I could finish then...

File: docs/components/concepts/workflow-patterns.md [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+
+1. How to find the right process instance? In Camunda, this is solved by a `message name` and a `correlation key` (e.g. `orderCanceled` and `order-42`).
+
+2. How to persist messages if a process instance is not yet ready to receive that message yet? In Camunda, this is solved by having an internal message store and a `time to live` attached to messages. This is related to [Workflow Pattern 24: Persistent Trigger](http://www.workflowpatterns.com/patterns/control/new/wcp24.php)
+
+You can find more information in [our documentation about messages](/docs/components/concepts/messages/).
+
+### Events from subprocesses
+
+Sometimes, you a subprocess need to communicate with its parent without ending the subprocess yet. BPMN allows this by an [escalation event](/docs/components/modeler/bpmn/bpmn-coverage/).
```
Comment (christinaausley) [inline docs/components/concepts/workflow-patterns.md:210]: Do you mean you need to communicate with a parent? Might just be a typo here.

File: docs/components/concepts/workflow-patterns.md [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+<div bpmn="workflow-patterns/subprocess-error.bpmn" callouts="errorEvent, errorEventSubprocess" />
+
+<span className="callout">1</span>
+
+This error event only catches errors from the subprocess "clearing". The idea here would be that in case of any clearing service not being available, the order is assumed cleared. Note that this example is mainly built for illustration, and does not necessarily mean this is the best way to solve this business requirement.
+
+<span className="callout">2</span>
+
+Alternatively, this error event subprocess is triggered whenever there is a fraud detected, independent where this error is raised.
```
Comment (christinaausley) [inline docs/components/concepts/workflow-patterns.md:258]: What do you mean by "independent where this error is raised"? Outside of where this error is raised?

**Review by christinaausley — CHANGES_REQUESTED** [review]
Two comments for clarification and a commit with my technical review to clean up some punctuation, typos, and clarification. Nice work here.

File: docs/components/concepts/workflow-patterns.md (hunk 1 above)
Comment (akeller) [reply docs/components/concepts/workflow-patterns.md:210]:
```suggestion
Sometimes, a subprocess needs to communicate with its parent process without ending the subprocess yet. BPMN allows this by an [escalation event](/docs/components/modeler/bpmn/bpmn-coverage/).
```

File: docs/components/concepts/workflow-patterns.md (hunk 2 above)
Comment (akeller) [reply docs/components/concepts/workflow-patterns.md:258]: It's not attached like a boundary event, it listens throughout the process, including other subprocesses. Maybe "independent of whether the error occurs in any subprocess or parent process"?

File: docs/components/concepts/workflow-patterns.md (hunk 1 above)
Comment (berndruecker) [reply docs/components/concepts/workflow-patterns.md:210]: Amara is completely right. And yes - typo

File: docs/components/concepts/workflow-patterns.md (hunk 2 above)
Comment (berndruecker) [reply docs/components/concepts/workflow-patterns.md]: Like this: https://github.com/camunda/camunda-platform-docs/pull/1006/commits/9aee9f96c83fa0247aa3e52506470ad1562951a3

**Comment by berndruecker** [comment]
Nice work @christinaausley - thanks in advance! Feel free to merge once the last comment is resolved.

File: docs/components/concepts/workflow-patterns.md (hunk 2 above)
Comment (christinaausley) [reply docs/components/concepts/workflow-patterns.md]: Excellent!

**Pull request merged**